presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -788,14 +800,45 @@ private synchronized void cacheLastDataCommitTimes(Map<String, Optional<Partitio
         }
     }
 
-    public synchronized void setPartitionLeases(MetastoreContext metastoreContext, String databaseName, String tableName, Map<String, String> partitionNameToLocation, Duration leaseDuration)
+    /**
+     * Set partition leases for a given table in the metastore.
+     *
+     * @param metastoreContext The context of the metastore
+     * @param databaseName The name of the database the table is in
+     * @param tableName The name of the table
+     * @param partitionNameToLocation A map of partition names to their location
+     * @param leaseDuration The duration of the lease
+     * @param leaseRenewalPeriod The period at which the lease should be renewed
+     */
+    public synchronized void setPartitionLeases(
+            MetastoreContext metastoreContext,
+            String databaseName,
+            String tableName,
+            Map<String, String> partitionNameToLocation,
+            Duration leaseDuration,
+            Duration leaseRenewalPeriod)
     {
         checkReadable();
         Table table = getTable(metastoreContext, databaseName, tableName)
                 .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
         boolean isPartitioned = table.getPartitionColumns() != null && !table.getPartitionColumns().isEmpty();
         if (table.getTableType().equals(MANAGED_TABLE) && isPartitioned && leaseDuration.toMillis() > 0) {
             delegate.setPartitionLeases(metastoreContext, databaseName, tableName, partitionNameToLocation, leaseDuration);
+            log.info("Setting partition lease for database %s, table %s, duration %s", databaseName, tableName, leaseDuration);
+
+            // If a lease renewal period is specified, schedule a task to periodically renew the lease
+            if (leaseRenewalPeriod.toMillis() > 0) {
+                Runnable runnable = () -> {
+                    log.info("Extending partition lease for database %s, table %s by %s", databaseName, tableName, leaseDuration);
+                    try {
+                        delegate.setPartitionLeases(metastoreContext, databaseName, tableName, partitionNameToLocation, leaseDuration);
```

Comment:
Should it be implemented as renew if necessary(i.e. reaching expiration of previous lease) rather than renewing every 10s or configured time. ?

If implemented this way, it makes sense to add in property description to chose a value which is close to lease duration to avoid unnecessary calls going to metastore.

Reply:
I personally find the concept of a "renewal period" to be more user-friendly as it is easy to understand. However, I am open to hearing other ideas and suggestions.

Agree that the idea is for example set `lease-duration` to `12h` and then `renewal-period` to `11h` to make sure it's always renewed just before expiration.

Reply:
I am updated the property descripting following your suggestion.